src/Stack/Build/Source.hs

```diff
@@ -405,6 +405,12 @@ loadLocalPackage boptsCli targets (name, (lpv, gpkg)) = do
                 Just STUnknown -> assert False mempty
                 Nothing -> mempty
 
+        filterSkippedComponents = Set.filter (not . (`elem` boptsSkipComponents bopts))
+
+        (exes, tests, benches) = (filterSkippedComponents exeCandidates,
```

Comment:
Yes, but that is because of the code in `Stack.Build.Execute.exesToBuild`. I added a comment with details. I think it's better if this part behaves consistently for all types of components.